src/main/kotlin/org/jmailen/gradle/kotlinter/pluginapplier/AndroidSourceSetApplier.kt

```diff
@@ -28,8 +27,10 @@ internal object AndroidSourceSetApplier : SourceSetApplier {
             .getOrNull()
             .orEmpty()
 
+        val emptyFileTree = project.files().asFileTree
```

Comment:
About the fix itself: `reduce` can't operate on empty collections, so I replaced it with `fold` and provide the initial value. 
The way I obtain initial value is inspired by this thread: https://discuss.gradle.org/t/re-how-to-create-an-empty-filetree/14362